content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade.md

```diff
@@ -30,7 +30,7 @@ The upgrade workflow at high level is the following:
 - You need to have a kubeadm Kubernetes cluster running version 1.16.0 or later.
 - [Swap must be disabled](https://serverfault.com/questions/684771/best-way-to-disable-swap-in-linux).
 - The cluster should use a static control plane and etcd pods or external etcd.
-- Make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.17.md) carefully.
+- Make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.17.md) carefully.
 - Make sure to back up any important components, such as app-level state stored in a database.
```

Comment:
@saschagrunert , Will this task page require an update of the link to the release notes for every new release?
May be the sentence should be removed?
Too bad the CHANGELOG file is not tagged as, `v1.17` instead of `1.17`.
There is a shortcode `latest-version` corresponding to each release (for example, v1.17).

```
Make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-{{< latest-version >}}.md)
```

Reply:
Unfortunately yes, but someone must have done this in the past, right? Do we have a chance to get the latest tag release automatically via the website generator? I'm not sure what the best way to continue here would be.

Reply:
@saschagrunert , I just searched for another shortcode. How about try the `release-branch` shortcode? Does this work:

```
Make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-{{< release-branch >}}.md)
```

Reply:
I think the release branch is prefixed with `release-`, like `release-1.17`. :thinking:

Reply:
Ugh, Ugh

Reply:
OK. Should probably log an issue about offering a release version parameter without the `v`.

Reply:
@kbhawkey -- Would suggest calling it `latest-semver`, as the `v` isn't syntactically correct for semantic versioning.
ref: https://semver.org/#is-v123-a-semantic-version

Reply:
Maybe we can use https://git.k8s.io/kubernetes/CHANGELOG/README.md where possible, as it links to all CHANGELOGs.

Reply:
Hello @justaugustus .
Your suggestions are all good.
It is easy to create a shortcode (latest-semver?) that returns the latest version of Kubernetes without the `v` prefix. There are a number of places where the `v<latest-version>` string is used in the docs, but I think it makes sense to offer a base version string. This way an author can provide a link to a specific CHANGELOG?

Reply:
@kbhawkey -- I'm not familiar enough with this repo to provide authoritative advice, but maybe we can have a `VERSION` file in each branch, which some code does manipulation on?

Reply:
Or make the determination on branch version based on the branch names?

Reply:
Hello @justaugustus .
Docs has a working system for versioning based on docs release (and branching ...).
For now, I think my proposed shortcode that returns the base release version string is 
sufficient for handling this case. The version/release setting changes per the fields set in config.toml.

Reply:
SGTM! Let me know what you need from us.

Reply:
Added a new `latest-release-notes` substitution inspired by https://github.com/kubernetes/website/pull/19110, WDYT?

Reply:
Hello @saschagrunert , A few thoughts about the proposed shortcode:
Is a special purpose shortcode needed or helpful to authors (to maintain the links to the external release notes)?
I see a number of places where the links point to different release versions, not just latest.
Should the full link to the latest CHANGELOG files be captured in a shortcode?
I searched for CHANGELOG in the docs repository and came up with a number (54 results in 35 files) of links in blog content as well as a few more in docs content.

Reply:
> Is a special purpose shortcode needed or helpful to authors (to maintain the links to the external release notes)?

I think so, this way we could easily link the latest changelog. :innocent: 

> I see a number of places where the links point to different release versions, not just latest.

Yeah I see that as a special case, like linking to a dedicated git file. I would not add dedicated short codes for that.

> Should the full link to the latest CHANGELOG files be captured in a shortcode?

I'm not sure about this point.

Reply:
OK.